packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -43,39 +51,44 @@ function isCustomElementRegistryAvailable() {
 }
 
 if (isCustomElementRegistryAvailable()) {
-    getCustomElement = customElements.get.bind(customElements);
-    defineCustomElement = customElements.define.bind(customElements);
-    HTMLElementConstructor = HTMLElement;
-} else {
-    const registry: Record<string, CustomElementConstructor> = create(null);
-    const reverseRegistry: WeakMap<CustomElementConstructor, string> = new WeakMap();
-
-    defineCustomElement = function define(name: string, ctor: CustomElementConstructor) {
-        if (name !== StringToLowerCase.call(name) || registry[name]) {
-            throw new TypeError(`Invalid Registration`);
-        }
-        registry[name] = ctor;
-        reverseRegistry.set(ctor, name);
-    };
-
-    getCustomElement = function get(name: string): CustomElementConstructor | undefined {
-        return registry[name];
+    const { get, define } = customElements;
```

Comment:
huge simplification here, just having to define `getUpgradableConstructor` for the two different code path (when custom element is available, vs when it is not available).